### PR TITLE
Add ability to disapprove outgoing emails

### PR DESCRIPTION
**Pull request by daredoes**
It seemed a bit weird that we can approve emails to go out, but then we're just stuck with them going out. This will add a disapprove button that will undo the actions of the approve button.

Thoughts and comments appreciated.

### CODE REVIEW

**Comment by binary1230** [comment]
my brain goes to: I think this would be cool *only* as an emergency safety valve and not as a switch used in normal operations.  

if we need this kind of fine grained control, individual categories can implement where needed, and we do this in various places.

**Comment by migetman9** [comment]
@RobRuana That makes sense. Like how you disapprove of my PR, not that you unapprove my PR.

File: uber/site_sections/emails.py [hunk 1]
```diff
@@ -93,6 +93,14 @@ def approve(self, session, ident):
         session.add(ApprovedEmail(ident=ident))
         raise HTTPRedirect('pending?message={}', 'Email approved and will be sent out shortly')
 
+    @csrf_protected
+    def unapprove(self, session, ident):
+        approved_email = session.query(ApprovedEmail).filter(ApprovedEmail.ident == ident).first()
+        if approved_email:
+            session.delete(approved_email)
+            raise HTTPRedirect('pending?message={}', 'Email disapproved and will no longer be sent out')
```
Comment (RobRuana) [inline uber/site_sections/emails.py]: ಠ_ಠ

Comment (earl7399) [inline uber/site_sections/emails.py]: Suggest: "Approval to send this e-mail has been rescinded, and no further copies will be sent." or something similar.

Comment (migetman9) [inline uber/site_sections/emails.py]: Thank you @earl7399 and I missed it I swear @RobRuana !

**Comment by kitsuta** [comment]
For the record, I'm fine with this, although a unit test to maintain coverage would be nice!

**Comment by RobRuana** [comment]
I am more inclined to merge this now that we've got the hotel booking info email in the system. Someone might accidentally send it early, and it would be nice to have some way to walk that action back.

Granted, they'd only have a 2.5 minute window (on average) to undo the "Approve", but it might be enough.

**Comment by kitsuta** [comment]
Given all this, is there a reason we haven't actually merged this PR?

**Comment by RobRuana** [comment]
I think this still needs some work to really indicate what it is doing to the user. Any emails that have already been sent can't be recalled. If we add this button we might want to also consider adding a 10 minute delay after approving an email before sending it.

**Comment by kitsuta** [comment]
That's fair! I agree that a delay on email approval would be nice.

**Comment by RobRuana** [comment]
Superseded by https://github.com/magfest/ubersystem/pull/3164

**Pull request closed without merging**